### PR TITLE
Return error for duplicate route/method

### PR DESCRIPTION
**Pull request by XieX**
This is the first Go code I've ever written, so please review with an extremely critical eye :P

Checks to see if a route+method already exist, and if it does returns an error.

Added a unit test for the duplicate route/method error.

### Testing

Verify unit test.

I added a duplicate route to `example/foo.go` and then made sure the unit test failed.

Alternatively/additionally you can add a duplicate route+method to gonfalon and run it against that to make sure the error happens (running it against gonfalon master detects no current dupes, which I take as a good sign).

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -981,6 +981,50 @@ func (p *parser) parseResponseComment(pkgPath, pkgName string, operation *Operat
...
+		return p.OpenAPI.Paths[route].Options
+	case http.MethodHead:
+		return p.OpenAPI.Paths[route].Head
+	case http.MethodTrace:
+		return p.OpenAPI.Paths[route].Trace
+	}
+	return nil
+}
+
+func (p *parser) setOperation(route string, method string, operation *OperationObject) error {
```
Comment (InTheCloudDan) [inline parser.go]: I'm not seeing how this method could return an error? The only return I see is the `return nil`. Getters/setters aren't as common in Go but might be valid here.

**Comment by InTheCloudDan** [comment]
@XieX you can take a look at an example unit test from `ld-find-code-refs` https://github.com/launchdarkly/ld-find-code-refs/blob/master/search/search_test.go#L280-L287

File: parser.go (hunk 1 above)
Comment (XieX) [reply parser.go]: I can't either, just wanted to follow what I presumed was a convention. I think I can just omit the return type and return nothing - would that be preferable?

As for creating a getter & setter I just thought it was clearer having the two switch statements near each other. But if you prefer I'm happy to leave the "setter" code where it was in `parseRouteComment` and change the "getter" to a function that just does the `nil` check and returns a boolean. That might be clearer in terms of intent.

**Comment by XieX** [comment]
> https://github.com/launchdarkly/ld-find-code-refs/blob/master/search/search_test.go#L280-L287

Thanks! So you can just invoke these methods directly? I had it in my head they were private and therefore not accessible via unit tests, but if that's not the case then we're in business 👍

File: parser.go (hunk 1 above)
Comment (InTheCloudDan) [reply parser.go]: we should probably leave the error handling in there and add it in the `default` block if we're never expecting to hit it. So in the chance it does happen at least we're alerted to it.

And agreed on the `bool` as a check might be clearer on intent but I'll leave that up to you on which path is best. I have not worked in this section of the code yet.

Comment (XieX) [reply parser.go]: Updated and added test 👍

**Review by InTheCloudDan — APPROVED** [review]

**Pull request merged**